src/components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,128 @@
+'use client';
+import { useMediaQuery, type CSSObject } from '@mui/material';
+import useId from '@mui/utils/useId';
+import type { ComponentType, PropsWithChildren, ReactNode } from 'react';
+import 'swiper/css';
+import 'swiper/css/pagination';
+import { Autoplay, FreeMode, Navigation, Pagination } from 'swiper/modules';
+import { Swiper, SwiperSlide } from 'swiper/react';
+import { SwiperOptions } from 'swiper/types';
+import { SectionTitle } from '../ProfilePage/ProfilePage.style';
+import { CarouselContainer, CarouselHeader } from './Carousel.style';
+import { DotsPaginationProps } from './DotsPagination';
+import { CarouselNavigationProps } from './Navigation';
+
+interface CarouselProps {
+  title?: string;
+  headerInfo?: ReactNode;
+  sx?: CSSObject;
+  CarouselNavigation?: ComponentType<CarouselNavigationProps>;
+  CarouselPagination?: ComponentType<DotsPaginationProps>;
```

Comment:
I think it would be helpful in the long run to have a `Carousel.types.ts` where we define some general types as
```
export interface CarouselNavigationBase {
  classNames: {
    navigationPrev: string;
    navigationNext: string;
  };
}

export interface PaginationBase {
  className: string;
}
```

and then use it here as 
```
  CarouselNavigation?: ComponentType<CarouselNavigationBase>;
  CarouselPagination?: ComponentType<CarouselPaginationBase>;
```

And the `DottedPaginationProps` can extend `CarouselPaginationBase`, while `CarouselNavigationProps` can extends the `CarouselNavigationBase`